pool.go

```diff
@@ -42,25 +42,10 @@ func max(a, b int) int {
 	return b
 }
 
-// Pool is a connection pool.
-type Pool interface {
-	Executor
-	Trier
-
-	// Acquire returns a connection from the pool
-	// blocking until a connection is available.
-	// Acquired connections must be released to the pool when no longer needed.
-	Acquire(context.Context) (PoolConn, error)
-
-	// Close closes all connections in the pool.
-	// Calling close blocks until all acquired connections have been released,
-	// and returns an error if called more than once.
-	Close() error
-}
-
-type pool struct {
+// Pool is a connection pool and is safe for concurrent use.
+type Pool struct {
 	isClosed bool
```

Comment:
This should be shared I think

Reply:
You're saying there should be a `pool.IsClosed()` method?

Reply:
It should be shared between copies of pools! Got it. Yes, good catch!

Reply:
changed